src/software/ai/hl/stp/play/dynamic_plays/support_tactic_scorers/support_tactic_candidate.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include <set>
+
+#include "software/ai/hl/stp/play/dynamic_plays/support_tactic_scorers/support_tactic_scorer.h"
+#include "software/ai/hl/stp/tactic/pass_receiver/pass_receiver_tactic.h"
+#include "software/ai/hl/stp/tactic/fake_pass_receiver/fake_pass_receiver_tactic.h"
+#include "software/ai/hl/stp/tactic/cherry_picker/cherry_picker_tactic.h"
+#include "software/ai/hl/stp/tactic/disrupter/disrupter_tactic.h"
+
+/**
+ * A SupportTacticCandidate is a "type wrapper" that makes a type of tactic
+ * eligible for scoring and assignment as a support tactic in a DynamicPlay.
+ *
+ * It acts as the visitee for a SupportTacticScorer, which operates under
+ * the visitor pattern.
+ */
+template<typename TSupportTactic>
+class SupportTacticCandidate 
+{
+    static_assert(std::is_base_of<Tactic, TSupportTactic>::value, 
+                  "TSupportTactic must derive from Tactic");
```

Comment:
why do we need this here?

Reply:
oh this is just asserting that the TSupportTactic is a Tactic. okay that's good.